tower-http/src/compression_utils.rs

```diff
@@ -308,3 +308,27 @@ where
 }
 
 pub(crate) const SENTINEL_ERROR_CODE: i32 = -837459418;
+
+#[allow(clippy::clippy::needless_bool)]
+pub(crate) fn supports_transparent_compression(headers: &HeaderMap) -> bool {
+    let content_type = if let Some(content_type) = content_type(headers) {
+        content_type
+    } else {
+        return true;
+    };
+
+    if content_type == "application/grpc" {
```

Comment:
probably want to do `starts_with` since sometimes grpc impls will tack on `+proto`/`+json`

Reply:
Good catch!